core/templates/components/forms/custom-forms-directives/image-uploader.directive.ts

```diff
@@ -51,19 +53,52 @@ angular.module('oppia').directive('imageUploader', [
             return 'This file is not recognized as an image.';
           }
 
-          if (!file.type.match('image/jpeg') &&
-            !file.type.match('image/gif') &&
-            !file.type.match('image/jpg') &&
-            !file.type.match('image/png') &&
-            !file.type.match('image/svg\\+xml')) {
-            return 'This image format is not supported.';
+          var imageTypeMapping = {
+            jpeg: {
+              format: 'image/jpeg',
+              fileType: /jp(e?)g$/,
+              fileExtension: /\.jp(e?)g$/,
+            },
+            jpg: {
+              format: 'image/jpg',
+              fileType: /jp(e?)g$/,
+              fileExtension: /\.jp(e?)g$/,
+            },
+            gif: {
+              format: 'image/gif',
+              fileType: /gif$/,
+              fileExtension: /\.gif$/,
+            },
+            png: {
+              format: 'image/png',
+              fileType: /png$/,
+              fileExtension: /\.png$/,
+            },
+            svg: {
+              format: 'image/svg\\+xml',
+              fileType: /svg\+xml$/,
+              fileExtension: /\.svg$/,
+            }
+          };
+          var imageHasInvalidFormat = true;
+          for (var i = 0; i < scope.getAllowedImageFormats().length; i++) {
+            var imageType = scope.getAllowedImageFormats()[i];
+            if (!imageTypeMapping.hasOwnProperty(imageType)) {
+              return 'Unknown image format in allowed image formats.';
+            }
+            if (file.type.match(imageTypeMapping[imageType].format)) {
+              imageHasInvalidFormat = false;
```

Comment:
What happens if, one of the "imageType" in the list has a valid format and the next one is invalid? According to this code, it looks like this boolean will still be set to false when it should be true.

Reply:
In that case the uploaded image is actually valid but the error will also be flagged.

Reply:
Ah, I see. Do you think it would make sense to mention the imageType that is invalid in the error string?

Reply:
Yes I think it would make sense to add the invalid imageType. Done.